### PR TITLE
Master

### PR DESCRIPTION
**Pull request by Serasidis**
- Added support to **Serial1** , **Serial2** and **Serial3** instances. **SerialUSB** port is not supported yet.
- Added the **STM32F103C8** variant.
- Is commented out the **#define true 0x1** and **#define false 0x0** from **wiring_constants.h** file because the 'true' and 'false' are also defined in
 `arm-none-eabi-gcc\4.8.3-2014q1\lib\gcc\arm-none-eabi\4.8.3\include\stdbool.h` file.  
If you compiler does not find these definitions, put them again in the source code.

### CODE REVIEW

File: HALMX/cores/mapleMX/wiring_constants.h [hunk 1]
```diff
@@ -30,8 +30,8 @@ extern "C"{
 #define OUTPUT 0x1
 #define INPUT_PULLUP 0x2
 
-#define true 0x1
-#define false 0x0
+//#define true 0x1
```
Comment (Serasidis) [inline HALMX/cores/mapleMX/wiring_constants.h]: The **true** and **false** are re-defined in `arm-none-eabi-gcc\4.8.3-2014q1\lib\gcc\arm-none-eabi\4.8.3\include\stdbool.h` file.

File: HALMX/cores/mapleMX/wiring_constants.h [hunk 2]
```diff
@@ -30,8 +30,8 @@ extern "C"{
...
+//#define false 0x0
```
Comment (Serasidis) [inline HALMX/cores/mapleMX/wiring_constants.h]: The true and false are re-defined in arm-none-eabi-gcc\4.8.3-2014q1\lib\gcc\arm-none-eabi\4.8.3\include\stdbool.h file.

**Comment by rogerclarkmelbourne** [comment]
At the moment I'm merging these on the server without testing them initially myself, as this is quicker.

If I find and problem I'll post a fix

**Comment by Serasidis** [comment]
ok Roger. I saw that HAL supports **DFU** and  **Virtual Port Com** . Do you think it's a good idea for me to add a DFU firmware bootloader ?

**Comment by rogerclarkmelbourne** [comment]
I think Virtual Com port is the next step, as libmaple has a Virtual com port (but using a different driver) for Serial

We should eventually be able to use the existing bootloader for DFU flashing the firmware

**Pull request merged**